src/main/java/org/haffson/adventofcode/days/day07/StepSorter.java

```diff
@@ -0,0 +1,56 @@
+package org.haffson.adventofcode.days.day07;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public class StepSorter {
+    @Nonnull
+    private final SortedSteps sortedSteps;
+
+    public StepSorter() {
+        this.sortedSteps = new SortedSteps();
+    }
+
+    @Nonnull
+    public String getSortedSteps(@Nonnull final List<Step> stepsInput) {
+        requireNonNull(stepsInput, "stepsInput");
+
+        final List<Step> stepsCopy = new ArrayList<>(stepsInput);
+
+        stepsInput.forEach(step -> addNextStepToResult(stepsCopy));
```

Comment:
`step` wird hier gar nicht genutzt. 
Das ist wieder der Fall mit dem schwer sichtbaren internen State (hier `sortedSteps`).

Wenn man `sortedSteps` in der Methode erzeugt und weiterreicht, macht man die Abhängigkeit sichtbar und die Methode thread-safe.

Reply:
Über den ungenutzten `step` haben wir im Slack schonmal versucht eine bessere Lösung zu finden.
Ich habe die `sortedSteps` jetzt in der Methode erstellt, aber das `step`-"Problem" bekomme ich nicht ganz gelöst.

Reply:
Ganz einfache Lösung dürfte sein:

    for (int i = 0; i < stepsInput.size(); i++) {
       addNextStepToResult(stepsCopy)
    }

Dann sieht man zumindest, dass es nur um die Anzahl der Aufrufe geht.

Reply:
Tatsächlich beinhaltete meine aller Erste Lösung eine `fori`-Schleife.
Die wurde aber moniert, da man nicht klar genug erkennen könne, warum geloopt wird bzw. mit einer Zählervariablen gerabeitet wird.
Ich solle beser eine `forEach`  nehmen :D

Reply:
Schlimm, wie hier beraten wird :joy: